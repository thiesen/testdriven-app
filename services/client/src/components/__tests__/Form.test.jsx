import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Form from '../Form';

const formData = {
  username: '',
  email: '',
  password: ''
};

describe('<Form/>', () => {
  it('renders registration form properly', () => {
    const component = <Form formType={'Register'} formData={formData}/>;
    const wrapper = shallow(component);

    const h1 = wrapper.find('h1');
    expect(h1.length).toBe(1);
    expect(h1.get(0).props.children).toEqual('Register');

    const formGroup = wrapper.find('.form-group');
    expect(formGroup.length).toBe(3);
    expect(formGroup.get(0).props.children.props.name).toEqual('username');
    expect(formGroup.get(0).props.children.props.value).toBe('');
  });

  it('renders login form properly', () => {
    const component = <Form formType={'Login'} formData={formData}/>;
    const wrapper = shallow(component);

    const h1 = wrapper.find('h1');
    expect(h1.length).toBe(1);
    expect(h1.get(0).props.children).toEqual('Login');

    const formGroup = wrapper.find('.form-group');
    expect(formGroup.length).toBe(2);
    expect(formGroup.get(0).props.children.props.name).toEqual('email');
    expect(formGroup.get(0).props.children.props.value).toEqual('');
  });

  test('Login form snapshot', () => {
    const component = <Form formType={'Login'} formData={formData} />;
    const tree = renderer.create(component).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('Register form snapshot', () => {
    const component = <Form formType={'Register'} formData={formData} />;
    const tree = renderer.create(component).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
