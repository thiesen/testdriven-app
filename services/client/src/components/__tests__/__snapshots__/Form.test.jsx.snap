// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form/> Login form snapshot 1`] = `
<div>
  <h1>
    Login
  </h1>
  <hr />
  <br />
  <form
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <input
        className="form-control input-lg"
        name="email"
        onChange={undefined}
        placeholder="Enter an email address"
        required={true}
        type="email"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <input
        className="form-control input-lg"
        name="password"
        onChange={undefined}
        placeholder="Enter a password"
        required={true}
        type="password"
        value=""
      />
    </div>
    <input
      className="btn btn-primary btn-lg btn-block"
      type="sumit"
      value="Submit"
    />
  </form>
</div>
`;

exports[`<Form/> Register form snapshot 1`] = `
<div>
  <h1>
    Register
  </h1>
  <hr />
  <br />
  <form
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <input
        className="form-control input-lg"
        name="username"
        onChange={undefined}
        placeholder="Enter a username"
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <input
        className="form-control input-lg"
        name="email"
        onChange={undefined}
        placeholder="Enter an email address"
        required={true}
        type="email"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <input
        className="form-control input-lg"
        name="password"
        onChange={undefined}
        placeholder="Enter a password"
        required={true}
        type="password"
        value=""
      />
    </div>
    <input
      className="btn btn-primary btn-lg btn-block"
      type="sumit"
      value="Submit"
    />
  </form>
</div>
`;
